### PR TITLE
Miscellaneous cleanups and unification 

### PR DESCRIPTION
**Pull request by alexanderConstantinescu**
**- What this PR does and why is it needed**

This PR does three things: 

- Commit `Cleanup egress code` cleans up some of the left overs from the last egress IP PR:  https://github.com/ovn-org/ovn-kubernetes/pull/1677 by re-structuring the code and addressing https://github.com/ovn-org/ovn-kubernetes/pull/1484#discussion_r462401005 / https://github.com/ovn-org/ovn-kubernetes/pull/1668#discussion_r507677137 / https://github.com/ovn-org/ovn-kubernetes/pull/1668#discussion_r507678581

- Commit `Unify retrieval of join GR IP addresses` unifies access to the gateway router IPs by using the new method `util.GetNodeGatewayRouterIPs`

- Commit `Move OVN prefix constants to util package` unifies access to the constants across the code base, something seen while writing `Unify retrieval of join GR IP addresses`

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by dcbw** [comment]
@alexanderConstantinescu can you rebase?

**Comment by alexanderConstantinescu** [comment]
@dcbw : rebased

File: go-controller/pkg/ovn/egressgw.go [hunk 1]
```diff
@@ -3,11 +3,12 @@ package ovn
 import (
 	"encoding/json"
 	"fmt"
-	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
-	utilnet "k8s.io/utils/net"
 	"net"
 	"strings"
 
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
```
Comment (trozet) [inline go-controller/pkg/ovn/egressgw.go:9]: nit: this should be grouped with "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"

File: go-controller/pkg/util/util.go [hunk 2]
```diff
@@ -27,6 +27,24 @@ const (
 	// access to local service
 	LocalNetworkName = "locnet"
 
+	// OVNClusterRouter is the name of the distributed router
```
Comment (trozet) [inline go-controller/pkg/util/util.go:30]: is util the right place to have all these constants? Seems a little odd to me, but I'm good with centralizing them to the same place that we currently use.

**Review by trozet — CHANGES_REQUESTED** [review]
minor nits

File: go-controller/pkg/util/util.go (hunk 2 above)
Comment (alexanderConstantinescu) [reply go-controller/pkg/util/util.go:30]: Not sure, it seems PR: https://github.com/ovn-org/ovn-kubernetes/pull/1754/commits/3b7906fa734192b01a53b7f3591b335348b416ec adds that to `pkg/config` 

I don't have a particularly strong opinion on the matter, but for this PR I would need it removed from `pkg/ovn/gateway.go` as to be able to use it from: https://github.com/ovn-org/ovn-kubernetes/blob/7ba0a081d49dfeb42323e22a8381d9f0c0b6906e/go-controller/pkg/util/net.go#L42 (the change introduced by my second commit) without introducing a cyclic dependency in the code

File: go-controller/pkg/ovn/egressgw.go (hunk 1 above)
Comment (alexanderConstantinescu) [reply go-controller/pkg/ovn/egressgw.go:9]: My IDE did that :innocent:

File: go-controller/pkg/util/util.go (hunk 2 above)
Comment (trozet) [reply go-controller/pkg/util/util.go:30]: I think config seems like a better place as long as it doesn't introduce a dependency problem. I'll  leave it @JacobTanenbaum to follow up on that

**Pull request merged**